TaskServiceInterview.cs

```diff
@@ -0,0 +1,448 @@
+                task.LockedByTimestamp = this.clock.Current;
+                isSuccessful = true;
+
+                task.AddBusinessEventGetLock(employeeInitials, this.clock.Current);
+                this.taskChangeNotificationServant.NotifySimpleEvent(task, TaskNotificationItem.OperationType.Update);
+            }
+
+            var result = this.mappingServant.Map<TaskEntity, TaskTryGetLockItem>(task);
+            result.IsSuccessful = isSuccessful;
+
+            return result;
+        }
+
+        public void ReleaseLock(Guid taskId, string employeeInitials)
+        {
+            //TODO 
+            var task = this.repository.Single(taskId, t => t.Classification, t => t.ExternalParty, t => t.Metadata, t => t.TaskEvents);
+
+            if (this.taskLockServant.EmployeeHasLock(task, employeeInitials))
+            {
+                task.ReleaseLock(employeeInitials, this.clock.Current);
+                this.taskChangeNotificationServant.NotifySimpleEvent(task, TaskNotificationItem.OperationType.Update);
+            }
+        }
+
+        public void ReleaseLocks(TaskReleaseLocksRequest request, string employeeInitials)
+        {
+            var tasks = this.taskFinder.FindActiveByLockedByInitials(employeeInitials)
+                .Where(x => request.TaskIds.Contains(x.Id)).ToList();
+            var now = this.clock.Current;
+            foreach (var task in tasks)
+            {
+                task.ReleaseLock(employeeInitials, this.clock.Current);
+                //external call
+                task.Deadline = this.businessCalendarClient.NextBusinessDay(now).Date;
+            }
+
+            this.NotifyAboutTaskChange(tasks, TaskNotificationItem.OperationType.Update);
+        }
+
+        public void ReleaseAllLocks(string employeeInitials)
+        {
+            var tasks = this.taskFinder.FindActiveByLockedByInitials(employeeInitials);
+
+            foreach (var task in tasks)
+            {
+                task.ReleaseLock(employeeInitials, this.clock.Current);
+                //external call
+                task.Deadline = this.businessCalendarClient.NextBusinessDay(this.clock.Current).Date;
+            }
+
+            this.NotifyAboutTaskChange(tasks, TaskNotificationItem.OperationType.Update);
+        }
+
+
+        public void FinishTask(TaskEntity task, string employeeInitials)
+        {
+            this.taskValidator.Validate(task, TaskValidationOperations.Finish);
+            task.Status = TaskStatus.Finished;
+            task.LockedByInitials = null;
+            task.LockedByTimestamp = null;
+
+            var now = this.clock.Current;
+
+            var lockEventTime = this.GetLockTime(task.Id, employeeInitials, now);
+
+            var finishBusinessEventData = new FinishBusinessEventData
+            {
+                TimeToCompleteTicks = (now - lockEventTime).Ticks,
+                DoneAfterDeadline = task.Deadline < this.clock.Current
```

Comment:
"Now" variable was created with the this.clock.Current value but here you are assigning the value in two different ways. Any reason to do this?